modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryWriterExImpl.java

```diff
@@ -325,8 +325,8 @@ else if (offsetByteCnt == BinaryUtils.OFFSET_2)
                 flags |= BinaryUtils.FLAG_HAS_RAW;
             }
             else {
-                finalSchemaId = 0;
```

Comment:
Objects which serialised by using raw writer only are affected by the very same issue. Looks like we don't need the `finalSchemaId` field at all